scripts/maintenance/NULL-assign

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/awk -f
```

Comment:
I see a problem with this: awk may not be in /usr/bin, or it may be named gawk. I'm not sure if all the scripts have been tested on non-GNU awk.
How about using a naming convention to define the interpreter and using the invoking script to locate it in a configure-like fashion?
I'm also okay with the main script bailing if it detects that awk is not GNU

Reply:
Aye, this is a side effect of making the scripts self-executable. As you may recall the original plan was to have a specific interpreter under the control of the main source-maintenance script. Now that possibility is gone we are left with this.

I am using /usr/bin/awk here because that is the tool name+location on our build system which will be the main user of these scripts. If we need to we can implement a system like the perl helper .in file building - but I would rather not complicate these things until we have a definite need.

Reply:
> the original plan was to have a specific interpreter under the control of the main source-maintenance script.

That original plan had the same kind of problem. The nuances and the solution space would have been naturally different, but the problem of using the right interpreter would not magically disappear just because the interpreter name is hard-coded in a different place.

> we are left with this.

That assertion feels misleading to me: There are many ways a plugin and/or its environment can control what is being executed.

> I would rather not complicate these things until we have a definite need.

FWIW, I agree with that we should not solve this (complicated) problem until the solution becomes cheaper than living with the problem.